### ISSUE TITLE
How do I disable the transmission of the message itself in the notification?

### ISSUE THREAD

**Issue by giterium** [nextcloud/spreed#2961]
Notifications go through apple and google servers. In many chats it is possible to disable the transfer of the username and the message itself in push.
Or tell me where in the code to make changes so that the message itself is not transmitted.
Thanks :)

**Comment by giterium** [comment]
And how does apple display them on the iphone screen if they are encrypted for it?

**Comment by giterium** [comment]
Apple itself issues a key that encrypts notifications, so they are unencrypted for it. Yes?

**Comment by giterium** [comment]
https://github.com/nextcloud/spreed/blob/master/lib/Notification/Notifier.php
248 and 249 lines
```
//$notification->setParsedMessage(str_replace($placeholders, $replacements, $message->getMessage()));
$notification->setParsedMessage(str_replace($placeholders, $replacements, 'fake message'));
//$notification->setRichMessage($message->getMessage(), $message->getMessageParameters());
                $notification->setRichMessage('fake message', $message->getMessageParameters());
```

**Comment by mario** [comment]
No, your device generates a key pair that encrypts the messages.

**Comment by nickvergessen** [comment]
> Apple itself issues a key that encrypts notifications, so they are unencrypted for it. Yes?

No it's nextcloud, and the nextcloud mobile apps decrypt the message

**Comment by nickvergessen** [comment]
And if you don't trust Apple on that level, you shouldn't trust it when opening the mobile app either, so you should get rid of the device all together.

**Comment by giterium** [comment]
I don't trust Apple to transfer data, and what's on my device legally belongs to me. If the app is turned off on ios it still participates in accepting the notification?

**Comment by giterium** [comment]
> 
> 
> No, your device generates a key pair that encrypts the messages.

It seems that the key is generated by the Apple or google notification server.

**Comment by nickvergessen** [comment]
The push token is generated by them. The key used to encrypt and decrypt the push notifications is generated by your device:

* Android https://github.com/nextcloud/talk-android/blob/6427e0bafddbfa0d730924073029e54907e17af7/app/src/main/java/com/nextcloud/talk/utils/PushUtils.kt#L157
* iOS https://github.com/nextcloud/talk-ios/blob/93c0616694b6fc89e8a3b901b8130b37f149971a/VideoCalls/NCSettingsController.m#L534

**Comment by nickvergessen** [comment]
Also see https://github.com/nextcloud/notifications/blob/master/docs/push-v2.md#pushed-notifications for more information.

We take privacy very seriously and we never sent unencrypted messages via 3rd party servers. Encryption was a pre-requirement before we added push notifications back then.

**Comment by giterium** [comment]
I didn't know about this
https://developer.apple.com/documentation/usernotifications/unnotificationserviceextension
Thanks for the explanation :)
